### PR TITLE
Add conceptFrom functions for ConceptAs and deprecate static from methods

### PR DESCRIPTION
**Pull request by woksin**
Can use the functions like this:

```
class SomeId extends ConceptAs<Guid, 'SomeId'> {
    constructor(id: Guid) {
        super(id, 'SomeId');   
    }
}

export function SomeIdFrom(id: Guid): SomeId {
    return conceptFrom(SomeId, id);
    // or return conceptFromFor(SomeId)(id);
}

const someId = SomeIdFrom(Guid.create());
```

### CODE REVIEW

**Review by joelhoisko — COMMENTED** [review]
Looks good, I added some tests for it too (and had to try it out myself 😋).

File: Source/concepts/ConceptAs.ts [hunk 1]
```diff
@@ -125,3 +129,10 @@ export class ConceptAs<T extends ConceptBase, U extends string> implements IEqua
         return this.value.toString();
     }
 }
+
+export function conceptFrom<C extends ConceptAs<T, U>, T extends ConceptBase, U extends string>(conceptConstructor: Constructor<ConceptAs<T, U>>, concept: T): C {
+    return new conceptConstructor(concept) as C;
+}
+export function conceptFromFor<C extends ConceptAs<T, U>, T extends ConceptBase, U extends string>(conceptConstructor: Constructor<ConceptAs<T, U>>): (concept: T) => C {
+    return (concept: T) => conceptFrom(conceptConstructor, concept) as C;
+}
```
Comment (joelhoisko) [inline Source/concepts/ConceptAs.ts:138]: I had a fat comment here about how to use this extra fancy `conceptFromFor()()` thingy but I accidentally closed the tab so in short:

Is there a reason you'd ever want to use the `()()` fancyness over the clear `conceptFrom(SomeId, id)` method?

Comment (joelhoisko) [reply Source/concepts/ConceptAs.ts:138]: Like you can add a static method like this to your concept class:
```typescript
class SomeId ...{
    /// ... class stuff ...

    static from: (concept: SomeId) {
        return conceptFromFor(SomeId);
    }
}

const someId = SomeId.from(Guid.create());
```

Or you could make reference on spot:
```typescript
const toSomeId = conceptFromFor(SomeId);
const someId = toSomeId(Guid.create());
```

Comment (woksin) [reply Source/concepts/ConceptAs.ts:138]: Yes, this was the scenarios I had in mind. Since we don't have a way to inherit this static method in a good way

**Review by joelhoisko — APPROVED** [review]

**Pull request merged**